src/2015-04-25-epidemiological-compartment-model.cpp

```diff
@@ -0,0 +1,240 @@
+/**
+ * @title Stochastic SIR Epidemiological Compartment Model
+ * @author Christian Gunning
+ * @license GPL (>= 2)
+ * @tags simulation parameter
+ * @summary Demonstrates a discrete, stochastic epidemiological 
+ *  using a tau-leap method.  This model 
+ *   takes a list of parameters and returns a data.frame
+ *   of simulation results.
+ *
+ * ### Introduction
+ *
+ * This post is a simple introduction to Rcpp for disease
+ * ecologists, epidemiologists, or dynamical systems 
+ * modelers - the sorts of folks who will benefit from a 
+ * simple but fully-working example.
+ * My intent is to provide a complete, 
+ * self-contained introduction to modeling with Rcpp.  
+ * My hope is that this model can be easily modified 
+ * to run any dynamical simulation that has 
+ * dependence on the previous time step 
+ * (and can therefore not be vectorized).
+ *
+ * This post uses a classic 
+ * Susceptible-Infected-Recovered (SIR)
+ * [epidemiological compartment model](https://en.wikipedia.org/wiki/Compartmental_models_in_epidemiology).
+ * Compartment models are simple, commonly-used dynamical
+ * systems models.  Here I demonstrate the
+ * [tau-leap method](https://en.wikipedia.org/wiki/Tau-leaping), where
+ * a discrete number of individuals move probabilistically 
+ * between compartments at fixed intervals in time.
+ * In this model, the wait-times within class 
+ * are exponentially distributed, and the number of
+ * transitions between states in a fixed time step are 
+ * Poisson distributed.
+ *
+ * This model is parameterized for the spread of measles
+ * in a closed population, 
+ * where the birth rate (nu) = death rate (mu).
+ * The transmission rate (beta) describes how frequently 
+ * susceptible (S) and infected (I) individuals come into 
+ * contact, and the recovery rate (gamma) describes the
+ * the average time an individual spends infected before
+ * recovering. 
+*/
+
+/** 
+ * ### C++ Code
+ * Note that cpp Functions must be marked with the following comment like for use in R
+ * // [[Rcpp::export]]
+*/
+
+#include <Rcpp.h>
+// Declare commonly used objects
```

Comment:
The part I like about namespace declarations is that it clearly shows the reader which parts of the API will be used.   I get that global imports are convenient.   Good programming books typically have a caveat of "don't do this yourself".  As this article is targetted towards novices,  I thought it useful to introduce the "declare only what you need" approach to show how it's done.

Would adding an explanatory comment assuage your concerns? :)